### PR TITLE
bpo-38823: Clean up refleaks in faulthandler initialization.

### PR DESCRIPTION
**Pull request by brandtbucher**
https://bugs.python.org/issue38823

### CODE REVIEW

**Comment by brandtbucher** [comment]
@vstinner

File: Modules/faulthandler.c [hunk 1]
```diff
@@ -1334,22 +1334,32 @@ PyInit_faulthandler(void)
 #ifdef MS_WINDOWS
     /* RaiseException() codes (prefixed by an underscore) */
     if (PyModule_AddIntConstant(m, "_EXCEPTION_ACCESS_VIOLATION",
-                                EXCEPTION_ACCESS_VIOLATION))
+                                EXCEPTION_ACCESS_VIOLATION)) {
+        Py_DECREF(m);
         return NULL;
```
Comment (vstinner) [inline Modules/faulthandler.c]: Hum, what do you think of adding a "error:" label which does the DECREF + return NULL?

Comment (brandtbucher) [reply Modules/faulthandler.c]: I actually prefer that. I just wanted to keep the diffs small for all of these.

Should I make the change?

Comment (brandtbucher) [reply Modules/faulthandler.c]: Done.

File: Modules/faulthandler.c [hunk 2]
```diff
@@ -1334,25 +1334,34 @@ PyInit_faulthandler(void)
...
-                                EXCEPTION_NONCONTINUABLE_EXCEPTION))
-        return NULL;
+                                EXCEPTION_NONCONTINUABLE_EXCEPTION)) {
+        goto error;
+    }
 #endif
 
     return m;
+
+error:
```
Comment (vstinner) [inline Modules/faulthandler.c:1363]: I expect a warning on unused label on Linux. This code should be surrounded by #ifdef MS_WINDOWS, no?

Comment (brandtbucher) [reply Modules/faulthandler.c:1363]: Thanks, fixed.

**Comment by vstinner** [comment]
Thanks. I took the liberty of rewriting the commit message.

**Pull request merged**